packages/core/sdk/src/auth/internal/CoreStitchAuth.ts

```diff
@@ -108,18 +121,28 @@ export default abstract class CoreStitchAuth<TStitchUser extends CoreStitchUser>
     this.authRoutes = authRoutes;
     this.storage = storage;
 
-    let info: AuthInfo | undefined;
+    let allUsersAuthInfo: { [key: string]: AuthInfo };
+    try {
+      allUsersAuthInfo = readCurrentUsersFromStorage(storage);
+    } catch(e) {
+      throw new StitchClientError(
+        StitchClientErrorCode.CouldNotLoadPersistedAuthInfo
+      );
+    }
+    this.allUsersAuthInfo = allUsersAuthInfo;
+
+    let activeUserAuthInfo: AuthInfo | undefined;
     try {
-      info = readFromStorage(storage);
+      activeUserAuthInfo = readActiveUserFromStorage(storage);
     } catch (e) {
       throw new StitchClientError(
         StitchClientErrorCode.CouldNotLoadPersistedAuthInfo
       );
     }
-    if (info === undefined) {
-      this.authInfo = AuthInfo.empty();
+    if (activeUserAuthInfo === undefined) {
+      this.activeUserAuthInfo = AuthInfo.empty();
     } else {
-      this.authInfo = info;
+      this.activeUserAuthInfo = activeUserAuthInfo;
```

Comment:
[q][minor] Didn't we agree that we wanted to preserve the order of the list? In which case, a map would no longer be the appropriate data structure?

Reply:
I'm just using a plain JS object, which I believe is ordered. the order did seem to be preserved in my tests.

In any case, after my next PR, the user-facing list will be sorted by `lastAuthActivity` and `userId`, so the on-device order will not necessarily matter. I'll probably do a lazy sorting approach so the list won't need to be re-sorted every time it is requested